forc-pkg/src/manifest/mod.rs

```diff
@@ -166,7 +166,7 @@ impl TryInto<WorkspaceManifestFile> for ManifestFile {
 type PatchMap = BTreeMap<String, Dependency>;
 
 /// A [PackageManifest] that was deserialized from a file at a particular path.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq)]
```

Comment:
Is there a specific reason why we are removing `Eq`? If not can we put them back, this makes the change breaking as it is, since this removes an implementation from a public type. 

Not in the sense that we need to bump sway repo to next minor but in the sense that forc-pkg as a library gets a breaking change in it's semver. But since we cannot version indivual crates currently, we would not be able to relay this information with the versioning.
 
If it was intentional it is fine but otherwise I think we should get it back

Reply:
It was intentional since the `toml::Value` does not derive the `Eq` trait impl.
Don't think we need a manual implementation for this.
Also took a look at the `cargo` implementation for the `metadata`; the struct also doesnt derive `Eq` probably due to this reason:
https://github.com/rust-lang/cargo/blob/010f171ca49ab4444806ff9568249d6a818c4103/crates/cargo-util-schemas/src/manifest/mod.rs#L120